tests/integration/security/jwt_test.go

```diff
@@ -409,61 +358,79 @@ func TestIngressRequestAuthentication(t *testing.T) {
 			}
-
 			applyPolicy("testdata/requestauthn/global-jwt.yaml.tmpl", rootNS{})
-			applyPolicy("testdata/requestauthn/ingress.yaml.tmpl", ns)
-
-			bSet := apps.B.Match(echo.Namespace(ns.Name()))
 
 			callCount := 1
 			if t.Clusters().IsMulticluster() {
 				// so we can validate all clusters are hit
 				callCount = util.CallsPerCluster * len(t.Clusters())
 			}
 
-			for _, cluster := range t.Clusters() {
-				t.NewSubTest(fmt.Sprintf("In %s", cluster.StableName())).Run(func(t framework.TestContext) {
-					a := apps.A.Match(echo.InCluster(cluster).And(echo.Namespace(apps.Namespace1.Name())))
-					// These test cases verify in-mesh traffic doesn't need tokens.
-					testCases := []authn.TestCase{
-						{
-							Name: "in-mesh-with-expired-token",
-							Request: connection.Checker{
-								From: a[0],
-								Options: echo.CallOptions{
-									Target:   bSet[0],
-									PortName: "http",
-									Scheme:   scheme.HTTP,
-									Headers: map[string][]string{
-										authHeaderKey: {"Bearer " + jwt.TokenExpired},
-									},
-									Count: callCount,
-								},
-								DestClusters: bSet.Clusters(),
+			t.NewSubTest("in-mesh-authn").Run(func(t framework.TestContext) {
+				// These test cases verify in-mesh traffic doesn't need tokens.
+				testCases := []authn.TestCase{
+					{
+						Name: "in-mesh-with-expired-token",
+						CallOpts: echo.CallOptions{
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Headers: map[string][]string{
+								authHeaderKey: {"Bearer " + jwt.TokenExpired},
 							},
-							ExpectResponseCode: response.StatusUnauthorized,
+							Count: callCount,
 						},
-						{
-							Name: "in-mesh-without-token",
-							Request: connection.Checker{
-								From: a[0],
-								Options: echo.CallOptions{
-									Target:   bSet[0],
-									PortName: "http",
-									Scheme:   scheme.HTTP,
-									Count:    callCount,
-								},
-								DestClusters: bSet.Clusters(),
-							},
-							ExpectResponseCode: response.StatusCodeOK,
+						ExpectResponseCode: response.StatusUnauthorized,
+					},
+					{
+						Name: "in-mesh-without-token",
+						CallOpts: echo.CallOptions{
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Count:    callCount,
 						},
-					}
-					for _, c := range testCases {
-						t.NewSubTest(c.Name).Run(func(t framework.TestContext) {
-							retry.UntilSuccessOrFail(t, c.CheckAuthn,
-								retry.Delay(250*time.Millisecond), retry.Timeout(30*time.Second))
+						ExpectResponseCode: response.StatusCodeOK,
+					},
+				}
+				for _, c := range testCases {
+					echotest.New(t, apps.All).
+						SetupForPair(func(t framework.TestContext, src, dst echo.Instances) error {
+							policy := yml.MustApplyNamespace(t, tmpl.MustEvaluate(
+								file.AsStringOrFail(t, "testdata/requestauthn/ingress.yaml.tmpl"),
+								map[string]string{
+									"Namespace": ns.Name(),
+									"dst":       dst[0].Config().Service,
```

Comment:
It looks like `testdata/requestauthn/ingress.yaml.tmpl` does not have the `dst` in the template, why passing it here?

Reply:
I reverted the change to `testdata/requestauthn/ingress.yaml.tmpl` back by mistake. This is templatized and has `dst` now.

Reply:
is the test still passing even without this change?

Reply:
The test passed without this change, but that conflicts with the purpose of the test. We have to pass dest service to `{{ .dst}}` in the `VirtualService` and let src call dst, to verify that jwt is not required because they are both in the mesh.
Without the change to the policy, when the dst is not `b`, (e.g. `a`->`vm`), `VirtualService` is still routing to `b`. That is not what the test is supposed to verify.